nixos/modules/virtualisation/oci-containers.nix

```diff
@@ -229,7 +229,7 @@ let
     dependsOn = map (x: "${cfg.backend}-${x}.service") container.dependsOn;
   in {
     wantedBy = [] ++ optional (container.autoStart) "multi-user.target";
-    after = lib.optionals (cfg.backend == "docker") [ "docker.service" "docker.socket" ] ++ dependsOn;
+    after = lib.optionals (cfg.backend == "docker") [ "docker.service" "docker.socket" ] ++ lib.optionals (container.imageFile != null) [ "network-online.target" ] ++ dependsOn;
```

Comment:
The dependency between imageFile and target is not obvious. Can you please add a comment.